### PR TITLE
Generate xml

### PR DESCRIPTION
**Pull request by jmgrady**
Implement generation of XML for from the `.docx` files

### CODE REVIEW

File: py-doc-to-xml/docx_to_xml/semantic_domain_xml.py [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+                node.Questions.add(
+                    ws=lang,
+                    index=i,
+                    question=domain_q.question,
+                    example_words=domain_q.words,
+                    example_sentences=domain_q.sentences,
+                )
+        else:
+            # The source document does not have the current domain so
+            # we copy the English text.
```
Comment (imnasnainaec) [inline py-doc-to-xml/docx_to_xml/semantic_domain_xml.py]: Let's throw a warning in this case.

File: py-doc-to-xml/docx_to_xml/semantic_domain_xml.py [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+            domain = new_domains[domain_abbr]
+            # The source document has the current domain
+            node.Abbreviation.add(ws=lang, value=domain.abbrev)
+            node.Name.add(ws=lang, value=domain.name)
+            node.Description.add(ws=lang, value=domain.description)
+            node_len = len(node.Questions.CmDomainQ)
+            new_len = len(domain.questions)
+            if node_len != new_len:
+                print(
+                    f"WARNING: Number of questions for {domain_abbr}, 'es' - {node_len}; {lang} - {new_len}",
```
Comment (imnasnainaec) [inline py-doc-to-xml/docx_to_xml/semantic_domain_xml.py]: Shouldn't this 'es' be 'en'?

File: py-doc-to-xml/docx_to_xml/semantic_domain_xml.py (hunk 1 above)
Comment (imnasnainaec) [reply py-doc-to-xml/docx_to_xml/semantic_domain_xml.py]: And let's only copy the Abbreviation from en... the other fields that still need to be translated should be left empty `ml` or `ta` elements.

**Review by imnasnainaec — CHANGES_REQUESTED** [review]

File: py-doc-to-xml/docx_to_xml/semantic_domain_xml.py (hunk 2 above)
Comment (jmgrady) [reply py-doc-to-xml/docx_to_xml/semantic_domain_xml.py]: Yes.  Fixed.

File: py-doc-to-xml/docx_to_xml/semantic_domain_xml.py (hunk 1 above)
Comment (jmgrady) [reply py-doc-to-xml/docx_to_xml/semantic_domain_xml.py]: Warning added.

Are you sure about leaving the fields blank?  In _The Combine_ UI any untranslated strings get the English version.  In crowd-in, will the English be visible?  What will the app do if there is a blank domain definition?  If we have the English, then at least the app is functional before translation is complete and the translated domain text can be viewed in _The Combine_.

Comment (jmgrady) [reply py-doc-to-xml/docx_to_xml/semantic_domain_xml.py]: If blank fields are what we want, do we want a single blank question if there are multiple questions or a blank question for each existing question in English?  Also, any of the options is straight-forward to implement and is fine with me.  I just want to confirm the choice.

Comment (imnasnainaec) [reply py-doc-to-xml/docx_to_xml/semantic_domain_xml.py]: I believe we want a blank for every question.

Comment (imnasnainaec) [reply py-doc-to-xml/docx_to_xml/semantic_domain_xml.py]: The the process of generating the JSON file from Crowdin uses the English when a translation is mission. For uploading to Crowdin, I think we want a blank field in the XML for everything that still needs to be translated.

Comment (jmgrady) [reply py-doc-to-xml/docx_to_xml/semantic_domain_xml.py]: Done.

File: py-doc-to-xml/docx_to_xml/util.py [hunk 3]
```diff
@@ -42,7 +42,7 @@ def split_question(s: str) -> Tuple[str, str]:
      1)
      (1)
     """
-    return split_line(s, pattern=r"\s*\(?(\d+)[).]?")
+    return split_line(s, pattern=r"\s*\(?([0-9]+)[).]?")
```
Comment (imnasnainaec) [inline py-doc-to-xml/docx_to_xml/util.py]: Can we change the `[).]?` to `[). ]` so it doesn't catch (for example) 8.4.4 question 4 example word "5:00".

Comment (jmgrady) [reply py-doc-to-xml/docx_to_xml/util.py]: I thought that there were cases where there was no whitespace between the question number and the question.  Even so, the question number should end in a '.' or ')' so I think we just want to get rid of the second '?' in the pattern.

Comment (imnasnainaec) [reply py-doc-to-xml/docx_to_xml/util.py]: Yes, that should be fine.

**Review by imnasnainaec — APPROVED** [review]

**Pull request merged**